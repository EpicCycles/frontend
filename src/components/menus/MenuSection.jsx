import React, {Fragment} from "react";
import {Link} from "react-router-dom";

const MenuSection = props => (
    <Fragment key={'menuSection' + props.sectionPos}>
        {props.sectionContents.map(section => {
            return (
                <Fragment key={'menuSection' + section.groupPos}>
                    <h2>{section.groupHeader}</h2>
                    <div className="">
                        {section.groupLinks.map(linkData => {
                            return <Link
                                key={`menuLink${linkData.linkNumber}`}
                                className="internal_link"
                                to={linkData.linkRoute ? linkData.linkRoute : linkData.linkURL}
                            >
                                {linkData.displayText}
                            </Link>
                        })}
                    </div>
                </Fragment>
            )
        })}
    </Fragment>
);
export default MenuSection;

