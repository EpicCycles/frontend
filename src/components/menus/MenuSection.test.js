import React from 'react';
import toJson from 'enzyme-to-json';
import MenuSection from './MenuSection';

test('it displays correctly', () => {
  const sectionContents = [
    {
      groupHeader: 'Customer',
      groupLinks: [
        { displayText: 'Find Customer', linkRoute: '/customer-search', linkNumber: 1 },
        { displayText: 'New Customer', linkRoute: '/customer', linkNumber: 2 },
      ],
    },
    {
      groupHeader: 'Core Data',
      groupLinks: [
        { displayText: 'Quote Sections', linkRoute: '/framework', linkNumber: 3 },
        { displayText: 'Brands', linkRoute: '/brands', linkNumber: 4 },
      ],
    },
  ];
  const component = shallow(<MenuSection sectionContents={sectionContents} />);
  expect(toJson(component)).toMatchSnapshot();
});
