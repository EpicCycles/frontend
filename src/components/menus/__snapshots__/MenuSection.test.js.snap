// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays correctly 1`] = `
<Fragment
  key="menuSectionundefined"
>
  <h2>
    Customer
  </h2>
  <div
    className=""
  >
    <Link
      className="internal_link"
      key="menuLink1"
      to="/customer-search"
    >
      Find Customer
    </Link>
    <Link
      className="internal_link"
      key="menuLink2"
      to="/customer"
    >
      New Customer
    </Link>
  </div>
  <h2>
    Core Data
  </h2>
  <div
    className=""
  >
    <Link
      className="internal_link"
      key="menuLink3"
      to="/framework"
    >
      Quote Sections
    </Link>
    <Link
      className="internal_link"
      key="menuLink4"
      to="/brands"
    >
      Brands
    </Link>
  </div>
</Fragment>
`;
